### PR TITLE
[mono] Add some strong types for a few variables, document some methods

### PR DESCRIPTION
**Pull request by migueldeicaza**
Some strong types that make it nicer to see variables in the debugger, plus
some inline API docs.

### CODE REVIEW

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -7076,11 +7076,11 @@ emit_init_local (MonoCompile *cfg, int local, MonoType *type, gboolean init)
 /*
  * inline_method:
  *
- *   Return the cost of inlining CMETHOD.
+ * Return the cost of inlining CMETHOD, or zero if it should not be inlined.
  */
 static int
 inline_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig, MonoInst **sp,
-			   guchar *ip, guint real_offset, gboolean inline_always)
+	       guchar *ip, guint real_offset, gboolean inline_always)
```
Comment (alexrp) [inline mono/mini/method-to-ir.c]: Indentation remains broken here as there's still a tab at the start of the line.

**Comment by vargaz** [comment]
build

**Comment by vargaz** [comment]
Looks ok.

File: mono/metadata/verify.h [hunk 2]
```diff
@@ -48,7 +49,7 @@ typedef struct {
 
 typedef struct {
 	MonoVerifyInfo info;
-	int8_t exception_type; /*should be one of MONO_EXCEPTION_* */
+	MonoExceptionType exception_type : 8; /*should be one of MONO_EXCEPTION_* */
```
Comment (evincarofautumn) [inline mono/metadata/verify.h]: Note that this is now unsigned with GCC and Clang, but signed with MSVC.

**Comment by vargaz** [comment]
build

**Pull request merged**